specifications/dpe-irot-profile/main.ocp

```diff
@@ -0,0 +1,1550 @@
+
+## Introduction
+
+This document describes a profile for implementation of DICE Protection
+Environment (DPE) [@{tcg-dpe}] on Integrated Roots of Trust (iRoTs). This profile is designed
+with the following properties to allow for implementation on a wide range of
+resource-constrained hardware:
+
+*   **Fixed-length command/response messages:** To avoid complex message-parsing in
+    minimalistic firmware environments, all commands and responses are fixed-size.
+*   **DPE instance running in the same SoC package as the client:** Some features
+    such as encrypted sessions are not supported because the client and DPE
+    implementation are assumed to reside within the same SoC.
+*   **Attestation-only:** This profile is only intended to provide the ability for
+    the DPE to provide attestation/signing functionality.
+
+This document defines multiple variants of the DPE iRoT profile:
+
+* `DPE_PROFILE_IROT_P256_SHA256`
+* `DPE_PROFILE_IROT_P384_SHA384`
+* `DPE_PROFILE_IROT_MIN_P256_SHA256`
+* `DPE_PROFILE_IROT_MIN_P384_SHA384`
+
+The profile name will be embedded as ASCII within the version field of the TCBInfo(s), enabling a verifier to confirm that the evidence originates from this profile.
+
+## Contexts
+
+### Handles and metadata
+
+Each call to DeriveContext creates a Context which is maintained internally to
+the DPE. Each context has a 16 byte handle which is used to reference the context
+externally.
+
+In this profile, all calls to DeriveContext include the following Input Data.
+See "DeriveContext ABI" for additional details of format and semantics.
+
+*   `INPUT_TYPE`: Type of the TCB component
+*   `INPUT_ALLOW_CA`: Whether contexts derived from this context allow
+    CertifyKey to create CA certificates
+*   `INPUT_ALLOW_X509`: Whether contexts derived from this context support
+    CertifyKey to create X.509-formatted certificates
+*   `INPUT_DATA`: Measurement of the TCB component
+
+This profile also supports localities. Each locality has a single plaintext
+session. This profile does not support encrypted sessions.
+
+DPE SHALL use the following inputs when deriving CDIs. Any changes to any
+of these inputs SHALL result in a different derived asymmetric key.
+
+*   Latest INPUT_DATA: The `INPUT_DATA` field from the most recent DeriveContext call
+    to involve this context.
+*   Cumulative INPUT_DATA: A combination of `INPUT_DATA` from all DeriveContext calls
+    involving this context.
+*   Type: The `INPUT_TYPE` field from the DeriveContext call which created this context
+*   Allow X509: Whether `INPUT_ALLOW_X509` was set when creating this context
+*   Allow CA: Whether `INPUT_ALLOW_CA` was set when creating this context
+*   Locality: The `LOCALITY` field from the DeriveContext call which created this context
+
+If `RECURSIVE` is not supported, an implementation MAY omit the "Cumulative" measurement
+when computing CDIs.
+
+See tcg.derive.kdf-cdi-* sections for additional details on how these fields are
```

Comment:
for all the formats, we should change tcg -> ocp. Since these are not defined by TCG.